src/libraries/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs

```diff
@@ -190,6 +190,7 @@ public partial class EventSource : System.IDisposable
         protected void WriteEvent(int eventId, long arg1, long arg2) { }
         protected void WriteEvent(int eventId, long arg1, long arg2, long arg3) { }
         protected void WriteEvent(int eventId, long arg1, string? arg2) { }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("EventSource Write Event is not safe due to type description data. Trimmer will not safely handle this case")]
```

Comment:
I think this message should be a little more "user focused". I don't think a caller of this API understands what "type description data" means.

How about 

```
[System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("EventSource WriteEvent will serialize the whole object graph. Trimmer will not safely handle this case because properties may be trimmed. This can be suppressed if the object is a primitive type")]
```